src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -95,13 +95,21 @@ public async Task StartContainersAsync(IExecutionContext executionContext, objec
             Version requiredDockerVersion = new Version(17, 12);
 #endif
 
-            if (dockerVersion.ServerVersion < requiredDockerVersion)
-            {
-                throw new NotSupportedException(StringUtil.Loc("MinRequiredDockerServerVersion", requiredDockerVersion, _dockerManger.DockerPath, dockerVersion.ServerVersion));
-            }
-            if (dockerVersion.ClientVersion < requiredDockerVersion)
+            var serverType = executionContext.Variables.Get(Constants.Variables.System.ServerType);
+            var agentOS = executionContext.Variables.Get(Constants.Variables.Agent.OS);
+
+            var checkDockerVersion = !(string.Equals(serverType, "Hosted", StringComparison.OrdinalIgnoreCase) &&  string.Equals(agentOS, "Linux", StringComparison.OrdinalIgnoreCase));
+
+            if (checkDockerVersion)
             {
-                throw new NotSupportedException(StringUtil.Loc("MinRequiredDockerClientVersion", requiredDockerVersion, _dockerManger.DockerPath, dockerVersion.ClientVersion));
+                if (dockerVersion.ServerVersion < requiredDockerVersion)
+                {
+                    throw new NotSupportedException(StringUtil.Loc("MinRequiredDockerServerVersion", requiredDockerVersion, _dockerManger.DockerPath, dockerVersion.ServerVersion));
+                }
+                if (dockerVersion.ClientVersion < requiredDockerVersion)
+                {
+                    throw new NotSupportedException(StringUtil.Loc("MinRequiredDockerClientVersion", requiredDockerVersion, _dockerManger.DockerPath, dockerVersion.ClientVersion));
+                }
             }
```

Comment:
Log the docker version and client version and also mention that we have skipped validation.